### DISCUSSION TITLE
Mapper ... has no properties with an explicit schema

### DISCUSSION THREAD

**Discussion by zopyx** [sqlalchemy/sqlalchemy#8490] [Usage Questions]
I have this classic one-to-many setup where an `InvoiceRequest` maps to multiple `Invoice` rows.
The code uses a configurable schema name.

```

SCHEMA = "foo"

class Invoice(Base):
    __table_args__ = {"schema": SCHEMA}
    __tablename__ = "invoices"

    id = Column(UUID(as_uuid=True), default=uuid.uuid4, primary_key=True)

    invoice_request_id = Column(UUID, ForeignKey(f"{SCHEMA}.invoice_requests.id") )
    invoice_request = relationship("InvoiceRequest", back_populates=f"{SCHEMA}.invoices")


class InvoiceRequest(Base):
    __table_args__ = {"schema": SCHEMA}
    __tablename__ = "invoice_requests"

    id = Column(UUID(as_uuid=True), default=uuid.uuid4, primary_key=True)

    invoices = relationship("Invoice", back_populates=f"{SCHEMA}.invoice_request")
```

When I try to add instantiate `InvoiceRequest`, I receive this error:

```
sqlalchemy.exc.InvalidRequestError: Mapper 'mapped class InvoiceRequest->invoice_requests' has no property 'public.invoices'
```

Removing all schema-related configurations from the code (aka `public` default schema), does not throw the error.
Any insights on how to configure schemas here properly?

Using SA 1.4, PG 13

**Comment by CaselIT** [comment]
Hi,

Not sure if `back_populates=f"{SCHEMA}.invoice_request"` comes from a tutorial or something like this, but that's definitely wrong.
there is no schema in back populates, since it refers to a property on the class the relationship points to.
If you remove it all should work
